seccomp/README.md

```diff
@@ -1,61 +1,225 @@
+
+   Check from the Linux command line   
+   ```
+   $ grep SECCOMP /boot/config-$(uname -r)
+   CONFIG_HAVE_ARCH_SECCOMP_FILTER=y
+   CONFIG_SECCOMP_FILTER=y
+   CONFIG_SECCOMP=y
+   ```
 
 ### Seccomp and Docker
 
-The original seccomp was what's now called "strict mode" where the only system calls allowed are `read`, `write`, `_exit` and `sigreturn`. This is not useful for docker containers because all docker programs need many other system calls just to initialize, load dynamic libraries, etc. The mode used by docker is called "filter mode". It uses BPF filters to control exactly which system calls to allow.
+Docker has used seccomp since version 1.10 of the Docker Engine.
 
-Docker has a JSON based DSL which allows defining seccomp profiles that compile down to seccomp filters. Profiles are passed to the `--run` command with the following flag:
+Docker uses seccomp in *filter mode* and has its own JSON-based DSL that allows you to define *profiles* that compile down to seccomp filters. When you run a container it gets the default seccomp profile unless you override this by passing the `--security-opt` flag to the `docker run` command.
 
-```
---security-opt seccomp=profile.json
-```
+The following example command starts an interactive container based off the Alpine image and starts a shell process. It also applies the `default.json` seccomp profile to it.
 
-The JSON file is sent from the Docker client to the Docker daemon and the docker daemon
-compiles it into a BPF program using a thin go wrapper around libseccomp: github.com/seccomp/libseccomp-golang.
+   ```
+   $ sudo docker run -it --rm --security-opt seccomp=default.json alpine sh ...
+   ```
 
-The best way to test the effect of your seccomp profiles is to add all capabilities and disable apparmor.
-The following example uses one of the profiles included in this guide to show how to prevent a program from making any system calls at all.
-This is a reliable way to make sure your seccomp profile is enforced.
+The above command sends the JSON file from the client to the daemon where it is compiled into a BPF program using a [thin Go wrapper around libseccomp](https://github.com/seccomp/libseccomp-golang).
 
-```
-$ docker run --rm -it --cap-add ALL --security-opt apparmor=unconfined --security-opt seccomp=seccomp-profiles/deny.json alpine sh
-```
+Docker seccomp profiles operate a whitelist approach that specifies allowed syscalls. All syscalls not on the whitelist are not permitted.
+
+Docker supports many security related technologies. It is possible for other security related technologies to interfere with your testing of seccomp profiles. For this reason, the best way to test the effect of seccomp profiles is to add all *capabilities* and disable *apparmor*. This gives you the confidence that any changes you implement through seccomp are not tainted by *capabilities* and *apparmor*.
```

Comment:
"tainted" seems like a bad word in a security context, like it might be a problem using seccomp with caps and apparmor. Maybe could be phrased as "This gives you the confidence the behaviour you see in the following steps is solely due to seccomp changes."